File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
import sys
sys.setrecursionlimit(10 ** 6) 


dp = {}

def fibo(n):
    if n <= 1:
        return n
    
    if n not in dp:
        dp[n] = (fibo(n - 2) + fibo(n - 1)) % 1234567
    
    return dp[n]

def solution(n):
    return fibo(n)
    

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    let fibo = [0 , 1]
    if (n <= 1) return fibo[n]
    
    for(let i = 2; i <= n; i++){
        fibo.push((fibo.at(-2) + fibo.at(-1)) % 1234567)
    }
    
    return fibo.at(-1)
}